README.md

```diff
@@ -0,0 +1,560 @@
+        super.onCreate(savedInstanceState);
+        
+        setupPush();
+        toastStartUpPushNotification();
+    }
+    
+    @Override
+    protected void onResume() {
+        super.onResume();
+        RegistrarManager.registerMainThreadHandler(this);
+        RegistrarManager.unregisterBackgroundThreadHandler(NotifyingHandler.instance);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        RegistrarManager.unregisterMainThreadHandler(this);
+        RegistrarManager.registerBackgroundThreadHandler(NotifyingHandler.instance);
+    }
+
+    @Override
+    public void onMessage(Context context, Bundle bundle) {
+        // display the message contained in the payload
+        Toast.makeText(getApplicationContext(),
+                bundle.getString(UnifiedPushMessage.ALERT_KEY), Toast.LENGTH_LONG).show();
+
+    }
+    
+    private void setupPush(){
+        PushApplication application = (PushApplication) getApplication();
+        PushRegistrar pushRegistrar = application.getPushRegistrar();
+        pushRegistrar.register(getApplicationContext(), new Callback<Void>() {
+            
+            @Override
+            public void onSuccess(Void data) {
+                Log.d(TAG, "Registration Succeeded");
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Log.e(TAG, e.getMessage(), e);
+                Toast.makeText(getApplicationContext(),
+                        "Ops, something is wrong :(", Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+    
+    public void toastStartUpPushNotification(){
+        if (getIntent() != null) {
+            Bundle bundle = getIntent().getExtras();
+            if (bundle != null && bundle.getString(UnifiedPushMessage.ALERT_KEY) != null) {
+                Toast.makeText(getApplicationContext(),
+                        bundle.getString(UnifiedPushMessage.ALERT_KEY), Toast.LENGTH_LONG).show();
+            }
+
+        }
+    }
+
+}
+```
+More information can be found at [docs.aerogear.org](https://docs.aerogear.org/aerogear/latest/push-notifications.html).
+
+### 4. Mobile-services parser
+The Mobile services parser is a helper class to parse the `mobile-services.json` file. 
+This file contains the connection details for the services used within the application.
+
+`IMobileService` has method stubs for configuration getters required to configure the `Apollo-Client`, `App-Auth` and `Unified push`. 
+These methods are:
+
+- `String getGraphqlServer()` = Return the URL for a GraphQL server.
+- `String getKIssuer()` = Returns the Issuer URL for AppAuth.
+- `String getKClientId()` = Returns the client Id for AppAuth 
+- `String getPushUrl()` = Returns the URL for the unified push server.
+- `String getPushVariantId()` = Returns the Variant ID for the unified push server.
+- `String getPushVariantSecret()` = Returns the Variant Secret for the unified push server.
```

Comment:
I don't really think we need to list the methods here. Again have them as code comments should be enough.